### DISCUSSION TITLE
Virtualization on iPad Pro with Apple Silicon M1

### DISCUSSION THREAD

**Discussion by kosmonavtika** [utmapp/UTM#3230] [UTM Help (iOS)]
The M1 chip in the latest Macs support virtualization instructions, and since the new iPad Pros (2021) use the same silicon, the instructions should be available for the new iPads as well.

I'd like to request implementation of virtualization support on M1 iPad Pros, if possible.

**Comment by osy** [comment]
Not possible until Apple lets us use hypervisor.framework.

**Reply by janik-io** [reply]
I recently came across this awesome article talking about using hardware-accelerated vms on iOS. In the article he stated the following about M1 based iPads:

> **These devices already have Hypervisor support unlocked in the kernel**
**so any jailbreak should work**, not just Fugu14
sign with the com.apple.private.hypervisor entitlement and include the decompiled Hypervisor.framework
https://worthdoingbadly.com/hv/

He also published his code to decompile the Hypervisor.framework: https://github.com/zhuowei/HvDecompile
and his modified version of UTM: https://github.com/zhuowei/UTM

Maybe this could help?

**Reply by jasoryeh** [reply]
Has anyone explored the above?

**Reply by Arna13** [reply]
From what I understand, a jailbreak is not needed but a vulnerability to be able to add entitlements to UTM is. 

I believe that people on 15.1 or earlier should be able to get this done with TrollStore, but I haven't tested it personally.

**Reply by osy** [reply]
I just pulled a 10 hour marathon and completed what zhuowei started. Windows 10 boots on my M1 iPad with trollstore. Update coming soon.

**Reply by jasoryeh** [reply]
That sounds promising, keep me posted. It would be nice if a jailbreak isn't needed as @Arna13 mentioned if at all possible and would make the iPad a beast of a productivity machine.

**Reply by janik-io** [reply]
@osy Wow! Awesome! 🥳 What iPad OS version do you use? I am stuck on iOS 15.5 😕 Nevertheless this sounds great.

**Reply by AlexCGul** [reply]
Hello! I was thinking of picking up a macbook, though I was thinking about virtualization and a few other things and came accross this article, I'm wondering how it worked for you?

**Reply by xerz-one** [reply]
@AC-Gul If you're lucky enough to find a 1/2TB iPad with an iPadOS old enough to apply a jailbreak or TrollStore on, it's superworth it. 512GB and lower can run Windows and Linux too, but if you want smooth graphics you might run out of RAM.

**Reply by AlexCGul** [reply]
Thankyou very much! That really helps!!!

**Reply by tifasoftware** [reply]
@AC-Gul If you wanting to primarily do Virtualization, I recommend a macbook for that, since you can run Linux and Windows, aswell as run the lastest macOS.
I recommend a macbook air with at least 512 gb or a macbook pro 13"

**Reply by AlexCGul** [reply]
My goal is to use it as a game dev station for school.
I would like to use blender and unity with it. I'm considering one of three options:
1. Get a m1 macbook
2. Get a windows laptop ( Something with a dedicated GPU )
3. Get an M1 ipad 

And while I'm aware that for my workload the macbook would simply be better,
I've got time to decide so should a method come up that works alright then I wouldn't mind making it work.

**Comment by xerz-one** [comment]
Out of curiosity, is this possible under jailbreak, even if it doesn't come out of the box on iPadOS?

**Reply by kmeisthax** [reply]
AFAIK, no, because to use hardware virtualization on ARM64 you need to be in EL2. iPads do not actually launch iPadOS in EL2, so no virtualization can be used whatsoever.

This won't change until and unless we either get an iBoot exploit or Apple decides that iPads are worthy of virtualization. I'm not sure which one is less likely...

**Reply by saagarjha** [reply]
iPads do launch the kernel in EL2, but they don’t enable the hypervisor.

**Reply by kmeisthax** [reply]
TIL. I wonder if it's possible to enable virtualization with a jailbreak tweak, then...

**Reply by osy** [reply]
Not without an iBoot exploit or PPL bypass.
